invenio_app_ils/ill/api.py

```diff
@@ -159,40 +186,152 @@ def create_loan(
             in user id will be used.
         :return: the PID and record loan
         """
+        rec = self.record
         tloc_pid = (
             transaction_location_pid
             or current_app.config["ILS_DEFAULT_LOCATION_PID"]
         )
         transaction_user_pid = transaction_user_pid or current_user.id
-        if arrow.get(self["loan_end_date"]) < arrow.now():
+        if arrow.get(end_date) < arrow.now():
             raise ILLError("The loan end date cannot be in the past.")
 
-        item_pid = dict(type=self._pid_type, value=self["pid"])
+        item_pid = dict(type=rec._pid_type, value=rec["pid"])
         pid, loan = checkout_loan(
             start_date=start_date,
-            end_date=self["loan_end_date"],
-            document_pid=self["document_pid"],
+            end_date=end_date,
+            document_pid=rec["document_pid"],
```

Comment:
~~Is it guaranteed that it will have `document_pid`?~~ as I understand the aforementioned record is an `item` so `document_pid` is guaranteed since it is required in the item schema.

Reply:
The record is a `BorrowingRequest` and yes you are right the `document_pid` must be there, enforced by the schema.